enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/catchup/tx/TxPullRequestHandler.java

```diff
@@ -97,7 +97,7 @@ protected void channelRead0( ChannelHandlerContext ctx, final TxPullRequest msg
                 String message = format( "Streamed transactions [%d--%d] to %s", firstTxId, txStream.lastTxId(), ctx.channel().remoteAddress() );
                 if ( f.isSuccess() )
                 {
-                    log.info( message );
+                    log.debug( message );
```

Comment:
You will be formatting the string every time, whether it's logged or not. Maybe assign the primitive message to a variable, use the log formatting for the success case, and String.format for the failure case.

Reply:
If the compiler wasn't able to optimise before, it should now!